index.ts

```diff
@@ -216,33 +217,37 @@ function suiteClassCallback(target: SuiteCtor, context: TestFunctions) {
                                 || (shouldOnlyParam && context.it.only)
                                 || context.it;
 
-                            applyTestFunc(testFunc, parametersTestName, method, [params], method.length <= 1);
+                            applyTestFunc(testFunc, parametersTestName, method, [params]);
                         }
                     });
                 } else {
                     const testFunc = (shouldSkip && context.it.skip)
                         || (shouldOnly && context.it.only)
                         || context.it;
 
-                    applyTestFunc(testFunc, testName, method, [], method.length === 0);
+                    applyTestFunc(testFunc, testName, method, []);
                 }
             }
         }
 
+        function isAsync(method: Function): boolean {
+
+            return RE_ASYNC.test(method.toString());
```

Comment:
👎 Can't this still use length? What does mocha use to check if a function is async?

Reply:
ups wonder why this is still in there...
mocha just uses the length.
but for our purposes, especially with parameter dependency injection and async tests, I think that we will need this and some convention on how to name the `done` parameter.
I will remove this for now.

Reply:
the problem is with @params, need to check whether the decorator is available...

Reply:
fixed

Reply:
🦅 eye... great!